http-01 wellknown support + ansible galaxy file structure :

DnR-iData opened the pull request:
+ http-01 wellknown support
+ dynamic ssl url var set
+ haproxy crt + key concatenation
+ ansible galaxy file structure

Commit:
http-01 wellknown support + ansible galaxy file structure :
+ http-01 wellknown support
+ dynamic ssl url var set
+ haproxy crt + key concatenation
+ ansible galaxy file structure

andrewhowdencom commented:
Holy shit this is awesome <3 I will review now.

andrewhowdencom commented on meta/main.yml:
```diff
@@ -0,0 +1,57 @@
+  # - CC-BY
+  license: license (GPLv2, CC-BY, etc)
+
+  min_ansible_version: 1.2
```
This requires some RSA stuff only available in 2.4+

https://github.com/sitewards/ansible-role-lets-encrypt/blob/master/tasks/lets_encrypt.yml#L43
http://docs.ansible.com/ansible/latest/openssl_csr_module.html

andrewhowdencom commented on tasks/main.yml:
```diff
@@ -1,4 +1,28 @@
 - include: "dependencies.yml"
+
+# Create the /.well-known/acme-challenge dir
+- include: "wellknow.yml"
```
this looks to be spelled incorrectly -- i think it's supposed to be `wellknown.yml`

andrewhowdencom replied:
Ah, I am incorrect - I see this now. Still, I think it makes more sense if they're both `wellknown` -- what do you think?

andrewhowdencom commented on tasks/providers/http-01/wellknown.yml:
```diff
@@ -0,0 +1,18 @@
+---
+#      # acme_data.challenge_data =>
```
What is the purpose of this comment? If I had to hazard a guess, I would imagine it was left here during debugging?

andrewhowdencom commented:
This is excellent, and the work is much appreciated. I have done a quick review, and a couple of things stand out. While I'd be happy to merge it as is (it does improve things), the marks on review would be thing I'd love to see tidied up before it hits main. 

It's super appreciated. <3

DnR-iData committed:
Correcting some typo error + clean up debug comments :
+ Changing min ansible version compatibility
+ Correcting some error (wellknow => wellknown)
- Removing debug comments

DnR-iData commented:
Hi,

Happy to see the feature is appreciated :)

I have done the corrections (typo, version).

Thank you 👍

aboritskiy commented:
Hi, it is appreciated, thank you for the contribution!

andrewhowdencom commented:
Heya @DnR-iData; 

I'm super sorry it took this long to get back to you. Unfortunately this was mine to Shepard into merging, and I simply lost it in the notification spam. 

It's merged now. Thanks again for your patience <3
